src/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -1199,7 +1267,44 @@ else if (split[1].equalsIgnoreCase("clear")) {
 							TownyMessaging.sendErrorMsg(player, e.getMessage());
 						}
 
-				} else if (split[0].equalsIgnoreCase("homeblock")) {
+				} else if (split[0].equalsIgnoreCase("banner")) {
+					
+					if (plugin.getTownyAPI().isMenusEnabled()==false) {
+						TownyMessaging.sendErrorMsg(player, "Command is Disabled. Server needs to be running on a Bukkit verison greater than 1.7.");
+						return;
+					}
+					
+					if (player.getItemInHand() == null || player.getItemInHand().getType() != Material.BANNER) {
+						TownyMessaging.sendErrorMsg(player, "You're not Holding a Banner!");
+						return;
+					}
+					
+					town.setBanner(BukkitTools.getBannerFromString(BukkitTools.getStringOfBanner(player.getItemInHand())));
+					TownyMessaging.sendMsg(player, "Set the Town's Banner Successfully! Give it a couple Minutes to update on /t list.");
+					
+				} else if (split[0].equalsIgnoreCase("warp")) {
+

```

Comment:
It might also make better sense to use /town warp [set/add/remove] [name] for the command's structure. Players can use /town warp ? and see the three sub commands and probably figure it out for themselves.